jscomp/others/js_array.ml

```diff
@@ -17,61 +17,110 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
 
 type 'a t = 'a array
 
-external toString : unit -> string  = "" [@@bs.send.pipe: 'a t as 'this]
-external toLocaleString : unit -> string  = "" [@@bs.send.pipe: 'a t as 'this]
-external concat : 'this -> 'this  = "" [@@bs.send.pipe: 'a t as 'this]
-external append : 'a -> 'this = "concat" [@@bs.send.pipe: 'a t as 'this]
+external isArray : 'a -> Js.boolean = "Array.isArray" [@@bs.val]
 
-external slice : int -> int -> 'this = "" [@@bs.send.pipe: 'a t as 'this]
-external slice_copy : unit -> 'this = "slice"[@@bs.send.pipe: 'a t as 'this]
-external slice_start : int -> 'this = "slice"[@@bs.send.pipe: 'a t as 'this]
+external length : 'a array -> int = "" [@@bs.get]
 
-external indexOf : 'a  -> int = "" [@@bs.send.pipe: 'a t as 'this]
-external indexOfFrom : 'a -> int ->  int = "indexOf" [@@bs.send.pipe: 'a t as 'this]
+
+(* Mutator functions
+*)
+external copyWithin : int -> 'this = "" [@@bs.send.pipe: 'a t as 'this]
+external copyWithinFrom : int -> int -> 'this = "copyWithin" [@@bs.send.pipe: 'a t as 'this]
+external copyWithinFromTo : int -> int -> int -> 'this = "copyWithin" [@@bs.send.pipe: 'a t as 'this]
+
+external fill : 'a -> 'this = "" [@@bs.send.pipe: 'a t as 'this]
+external fillFrom : 'a -> int -> 'this = "" [@@bs.send.pipe: 'a t as 'this]
+external fillFromTo : 'a -> int -> int -> 'this = "" [@@bs.send.pipe: 'a t as 'this]
+
+(** https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push *)
+external pop : 'a array -> 'a Js.undefined = "" [@@bs.send]
```

Comment:
I'm thinking these should have pipe form too, but was hesitant to make the breaking change.